core/src/main/scala/org/apache/spark/deploy/worker/DriverWrapper.scala

```diff
@@ -66,4 +77,68 @@ object DriverWrapper {
         System.exit(-1)
     }
   }
+
+  // R or Python are not supported in cluster mode so download the jars to the driver side
+  private def setupDependencies(loader: MutableURLClassLoader, userJar: String): Unit = {
+    val packagesExclusions = sys.props.get("spark.jars.excludes").orNull
```

Comment:
Hmm, this is too much copy & paste for my liking. Could you instead refactor this code into a helper class or object (e.g. `IvytUtils`)?